modules/conversion-center/classes/render/core-render.php

```diff
@@ -16,8 +15,42 @@
 class Core_Render extends Render_Base {
 
 	public function render(): void {
+		$layout_props_full_height          = $this->settings['advanced_layout_full_screen_height'] ?? '';
+		$layout_props_full_height_controls = $this->settings['advanced_layout_full_screen_height_controls'] ?? '';
+		$layout_props_full_width           = $this->settings['advanced_layout_full_width_custom'] ?? '';
+		$layout_props_show_border          = $this->settings['background_show_border'] ?? '';
+		$custom_classes                    = $this->settings['advanced_custom_css_classes'] ?? '';
+
+		$layout_classnames = 'e-link-in-bio';
+
+		if ( 'yes' === $layout_props_show_border ) {
+			$layout_classnames .= ' has-border';
+		}
+
+		if ( 'yes' === $layout_props_full_height ) {
+			$layout_classnames .= ' is-full-height';
+		}
+
+		if ( 'yes' === $layout_props_full_width ) {
+			$layout_classnames .= ' is-full-width';
+		}
+
+		if ( ! empty( $layout_props_full_height_controls ) ) {
+			foreach ( $layout_props_full_height_controls as $breakpoint ) {
+				$layout_classnames .= ' is-full-height-' . $breakpoint;
+			}
+		}
+
+		if ( $custom_classes ) {
+			$layout_classnames .= ' ' . $custom_classes;
+		}
```

Comment:
```suggestion
		$layout_classnames = $this->get_layout_classnames();
```
Extract

Reply:
Maybe use a `prefix_class`? We generally try to avoid then because they add so much classes to the DOM, but it looks like you are building those classes anyway.
Is there **any** way to avoid all these class names and just use media queries?

Reply:
@louiswol94 these classes are coming from switch controls that allow to toggle borders and full width/height.
For the full height we are already using media queries, but for the border and full width we are not depending on the viewport.
The reason why we had to go this route is that it looks like the Switcher control doesn't allow adding CSS properties to the classes.
Can you advise on an alternative approach to this?

Reply:
@willhowat If a prefix_class does not work then this is fine. Please just extract this code to a separate private function. Thanks.

Reply:
@louiswol94 refactored code and extracted the method